### PR TITLE
OWLS-76689 - SOA:ADMINSERVER TAKING TIME TO RESTART USING SERVERSTARTPOLICY(NEVER TO IF_NEEDED) ON OKE CLUSTER 

### PR DESCRIPTION
**Pull request by alai8**
Minor changes to preCreateServerServices behavior by removing precreated services for managed servers when both admin server and all managed servers in the cluster are being brought down.                      

Issue fixed (OWLS-76689):
In SOA domains, the startup of admin server involves JNDI lookup of managed servers, restarting the admin server was found to be very slow, when the preCreateServerServices feature was enabled. This is due to each managed server host name was able to resolve due to the existence of precreated server services, so the JNDI lookup will attempt to connect to each managed server until the configured timeout value in the JNDI lookup, since the managed servers are not running. With this change, each lookup of the managed server will result in a UnknownHostException, and failed fast, which was the same behavior as when the admin server was started up the first time when the domain was first started.

Filer verified that the change reduced the admin server startup time from 22 minutes to around 2 minutes in the restart scenario. Filer also verified that the
issue in OWLS-75155, which was why the preCreateServerService feature was introduced, did not regress with this change.

jenkins http://wls-jenkins.us.oracle.com/job/weblogic-kubernetes-operator-javatest/2990/

### CODE REVIEW

File: operator/src/test/java/oracle/kubernetes/operator/steps/ManagedServersUpStepTest.java [hunk 1]
```diff
@@ -477,6 +489,46 @@ public void whenShuttingDown_allowAdminServerNameInListOfServers() {
...
   }
 
+  @Test
+  public void whenClusterStartupDefinedWithPreCreateServerService_addAllToServers() {
+    configureCluster("cluster1").withPrecreateServerService(true);
+    addWlsCluster("cluster1", "ms1", "ms2");
+
+    invokeStep();
+
+    assertThat(getServers(),
```
Comment (russgold) [inline operator/src/test/java/oracle/kubernetes/operator/steps/ManagedServersUpStepTest.java]: Not sure why you have this on multiple lines. 

allOf( hasItem("ms1"), hasItem("ms2") )

doesn't take up that much horizontal space, and seems a bit easier to read, I think. In fact, I suspect that the entire assert would fit comfortably on a single line.

File: operator/src/main/java/oracle/kubernetes/operator/steps/ManagedServersUpStep.java [hunk 2]
```diff
@@ -156,7 +156,7 @@ void addServerIfNeeded(@Nonnull WlsServerConfig serverConfig, WlsClusterConfig c
         servers.add(serverName);
         addStartupInfo(new ServerStartupInfo(serverConfig, clusterName, server));
         addToCluster(clusterName);
-      } else if (server.isPrecreateServerService()) {
+      } else if ((!domain.isShuttingDown() || server.shouldStart(0)) && server.isPrecreateServerService()) {
```
Comment (russgold) [inline operator/src/main/java/oracle/kubernetes/operator/steps/ManagedServersUpStep.java]: This condition is a bit complex. What is the intent? Can you describe it in a simple phrase that could be a method name?

File: operator/src/test/java/oracle/kubernetes/operator/steps/ManagedServersUpStepTest.java (hunk 1 above)
Comment (russgold) [reply operator/src/test/java/oracle/kubernetes/operator/steps/ManagedServersUpStepTest.java]: I don't see a change or response to this comment.

Comment (alai8) [reply operator/src/test/java/oracle/kubernetes/operator/steps/ManagedServersUpStepTest.java]: Oops. I missed some occurrences. Should be fixed now. Thanks!

**Pull request merged**